quarkus-extension/engine/runtime/src/main/java/org/camunda/bpm/quarkus/engine/extension/impl/CamundaEngineRecorder.java

```diff
@@ -67,13 +68,17 @@ public void configureProcessEngineCdiBeans(BeanContainer beanContainer) {
 
       // create a non-bean ManagedExecutor instance. This instance
       // delegates tasks to the Quarkus core Executor/thread pool.
-      ManagedExecutor managedExecutor = ManagedExecutor.builder()
+      ManagedExecutor managedExecutor = SmallRyeManagedExecutor.builder()
           .maxQueued(queueSize)
           .maxAsync(maxPoolSize)
+          .withNewExecutorService()
           .build();
+      ManagedJobExecutor quarkusJobExecutor = new ManagedJobExecutor(managedExecutor);
+
+      // apply job executor configuration properties
+      PropertyHelper
+          .applyProperties(quarkusJobExecutor, camundaEngineConfig.jobExecutor.configuration, PropertyHelper.KEBAB_CASE);
 
-      ManagedJobExecutor quarkusJobExecutor
-          = new ManagedJobExecutor(managedExecutor, camundaEngineConfig);
       configuration.setJobExecutor(quarkusJobExecutor);
```

Comment:
Can we encapsulate the whole job executor configuration logic into a dedicated method to ensure the readability of `createProcessEngineConfiguration` and make the method a bit smaller? Similarly to how I did it here:
https://github.com/camunda/camunda-bpm-platform/blob/e8c8af9d46ba88d9d77862c16073f162d8c4f869/quarkus-extension/engine/runtime/src/main/java/org/camunda/bpm/quarkus/engine/extension/impl/CamundaEngineRecorder.java#L87-L110

Reply:
Nice idea!